config.md

```diff
@@ -0,0 +1,164 @@
+# The js-ipfs config file
+
+The js-ipfs config file is a JSON document located in the root directory of the js-ipfs repository.
+
+## Table of Contents
+
+- [`Addresses`](#addresses)
+- [`Bootstrap`](#bootstrap)
+- [`Datastore`](#datastore)
+- [`Discovery`](#discovery)
+- [`Identity`](#identity)
+- [`Swarm`](#swarm)
+
+
+## `Addresses`
+Contains information about various listener addresses to be used by this node.
+
+- `API`
+The IPFS daemon exposes an HTTP API that allows to control the node and
+run the same commands as you can do from the command line. It is defined
+on the [HTTP API Spec](https://docs.ipfs.io/reference/api/http).
+
+Multiaddr describing the address to serve the local HTTP API on.
+
+Default: `/ip4/127.0.0.1/tcp/5002`
+
+- `Gateway`
+A gateway is exposed by the IPFS daemon, which allows an easy way to
+access content from IPFS, using an IPFS path.
+
+Multiaddr describing the address to serve the local gateway on.
+
+Default: `/ip4/127.0.0.1/tcp/9090`
+
+- `Swarm`
+Array of multiaddrs describing which addresses to listen on for p2p swarm connections.
+
+Default:
+```json
+[
+  "/ip4/0.0.0.0/tcp/4002",
+  "/ip4/127.0.0.1/tcp/4003/ws"
+]
+```
+
+## `Bootstrap`
+Bootstrap is an array of multiaddrs of trusted nodes to connect to in order to
+initiate a connection to the network.
+
+## `Datastore`
+Contains information related to the construction and operation of the on-disk
+storage system.
+
+- `Spec`
+Spec defines the structure of the ipfs datastore. It is a composable structure, where each datastore is represented by a json object. Datastores can wrap other datastores to provide extra functionality (eg metrics, logging, or caching).
+
+This can be changed manually, however, if you make any changes that require a different on-disk structure, you will need to run the [ipfs-ds-convert tool](https://github.com/ipfs/ipfs-ds-convert) to migrate data into the new structures.
+
+For more information on possible values for this configuration option, see docs/datastores.md
```

Comment:
Does `docs/datastores.md` exist?

Reply:
No, I will remove that for now. I am not really familiar with the datastores side of things, and I prefer to not block this PR now, and I can get it done on a next PR.